src/main/scala/com.snowplowanalytics.dataflow/streaming/StreamingCounts.scala

```diff
@@ -0,0 +1,181 @@
+import com.google.cloud.dataflow.sdk.options.DataflowPipelineOptions
+import com.google.cloud.dataflow.sdk.options.PipelineOptionsFactory
+import com.google.cloud.dataflow.sdk.runners.DataflowPipelineRunner
+import com.google.cloud.dataflow.sdk.transforms.DoFn
+import com.google.cloud.dataflow.sdk.transforms.ParDo
+import com.google.cloud.dataflow.sdk.transforms.PTransform
+import com.google.cloud.dataflow.sdk.transforms.GroupByKey
+import com.google.cloud.dataflow.sdk.transforms.windowing.FixedWindows
+import com.google.cloud.dataflow.sdk.transforms.windowing.Window
+import com.google.cloud.dataflow.sdk.values.KV
+import com.google.cloud.dataflow.sdk.values.PDone
+import com.google.cloud.dataflow.sdk.values.PCollection
+
+// Bigtable
+import com.google.cloud.bigtable.dataflow.AbstractCloudBigtableTableDoFn
+import com.google.cloud.bigtable.dataflow.CloudBigtableTableConfiguration
+import com.google.cloud.bigtable.dataflow.CloudBigtableTableConfiguration.Builder
+
+//HBase
+import org.apache.hadoop.hbase.client.Connection
+
+// This project
+import storage.BigtableUtils
+
+ /**
+   * transform applied to count the events per type in each bucket and 
+   * store the counts in Bigtable
+   */
+class StreamingCounts(config: CloudBigtableTableConfiguration, tableName: String, columnFamily: String)
+  extends PTransform[PCollection[KV[String, JIterable[String]]], PDone] {
+
+  override def apply(counts : PCollection[KV[String, JIterable[String]]]) : PDone = {
+  counts.apply(ParDo.named("StoreEventCounts").of(
+    StreamingCounts.storeEventCounts(config, tableName, columnFamily)))
+    PDone.in(counts.getPipeline)
+  }
+}
+
+object StreamingCounts {
+  /**
+   * Private function to set up Bigtable
+   *
+   * @param config The configuration for our job using StreamingCountsConfig.scala
+   * @return A Cloud Bigtable configuration instance with the options provided in config
+   */
+  private def setupBigtable(config: StreamingCountsConfig): CloudBigtableTableConfiguration =
+    new CloudBigtableTableConfiguration.Builder()
+      .withProjectId(config.projectId)
+      .withInstanceId(config.instanceId)
+      .withTableId(config.tableName)
+      .build()
+
+  /**
+   * Private function to set up Dataflow
+   *
+   * @param config The configuration for our job using StreamingCountsConfig.scala
+   * @param bigtableConfig The Bigtable config obtained from running setupBigtable
+   * @return a Pipeline instance, to which the transforms will be applied
+   */
+  private def setupDataflow(config: StreamingCountsConfig, 
+    bigtableConfig: CloudBigtableTableConfiguration): Pipeline = {
+ 
+    val options = PipelineOptionsFactory
+      .as(classOf[DataflowPipelineOptions])
+ 
+    options.setRunner(classOf[DataflowPipelineRunner])
+    options.setProject(config.projectId)
+    options.setStagingLocation(config.stagingLocation)
+    options.setStreaming(true)
+    Pipeline.create(options)
+  }
+
+  /**
+   * function applied in the "map" phase
```

Comment:
I think it's worth mentioning that you discard elements that are not properly formatted